src/Pos/Worker.hs

```diff
@@ -58,6 +60,12 @@ allWorkers NodeResources {..} = mconcatPair
     , wrap' "delegation" $ dlgWorkers
     , wrap' "slotting"   $ (properSlottingWorkers, mempty)
 
+    , case ncTopology ncNetworkConfig of
+        TopologyBehindNAT dnsDomains ->
+          wrap' "subscription" $ subscriptionWorkers ncNetworkConfig dnsDomains
+        _otherwise ->
+          mempty -- No subscription worker required
```

Comment:
Yes, but the KademliaParams aren't part of the topology. I had previously added a TODO that they ought to be added to the appropriate constructors, but you removed it and added it as a separate type so I assumed that that wouldn't work so well. Anyway feel free to modify as you think is appropriate, I'm out of time for today :)